tests/bdd/features/helm_chart/test_install.py

```diff
@@ -0,0 +1,251 @@
+    root_dir = os.environ["ROOT_DIR"]
+    helm_chart_path = root_dir + "/chart"
+
+    kube_config.load_kube_config()
+    core_v1 = kube_client.CoreV1Api()
+    # Listing the cluster nodes
+    node_list = core_v1.list_node()
+    # Checking to see if cluster can accommodate more than one, and less than or equal to 3 replicas of etcd.
+    usable_etcd_nodes = 0
+    for node in node_list.items:
+        # Skipping control-plane nodes
+        if is_k8s_control_plane_node(node):
+            continue
+
+        usable_etcd_nodes += 1
+
+        if usable_etcd_nodes == 3:
+            # If we're here, then clearly there's enough nodes to house a 3-replica etcd. We can break
+            # and no further counting is required.
+            break
+
+    if usable_etcd_nodes == 0:
+        raise ValueError("No kubernetes worker nodes in cluster")
+
+    # Install helm release.
+    result = run(["helm",
+                  "install",
+                  # Helm release name
+                  pytest.helm_release,
+                  # Path to the helm chart in this repository
+                  helm_chart_path,
+                  "-n",
+                  # Kubernetes namespace
+                  pytest.helm_namespace,
+                  "--dependency-update",
+                  "--set",
+                  # Disable callhome and eventing.
+                  "obs.callhome.enabled=false",
+                  "--set",
+                  # Number of etcd StatefulSet replicas
+                  "etcd.replicaCount=" + str(usable_etcd_nodes),
+                  # Wait no more than 15 minutes for all Pods, PVCs, Services to be ready.
+                  "--wait",
+                  "--timeout",
+                  "15m"
+                  ], check=True)
+
+
+@then('one agent-core Kubernetes Deployment is Available')
+def one_agentcore_kubernetes_deployment_is_available():
+    """one agent-core Kubernetes Deployment is Available."""
+    def predicate() -> bool:
+        return deployment_is_available(pytest.helm_release + "-agent-core", pytest.helm_namespace)
+
+    assert retry_predicate(predicate, pytest.max_retries, pytest.retry_seconds)
+
+
+@then('one api-rest Kubernetes Deployment is Available')
+def one_apirest_kubernetes_deployment_is_available():
+    """one api-rest Kubernetes Deployment is Available."""
+    def predicate() -> bool:
+        return deployment_is_available(pytest.helm_release + "-api-rest", pytest.helm_namespace)
+
+    assert retry_predicate(predicate, pytest.max_retries, pytest.retry_seconds)
+
+
+@then('one csi-controller Kubernetes Deployment is Available')
+def one_csicontroller_kubernetes_deployment_is_available():
+    """one csi-controller Kubernetes Deployment is Available."""
+    def predicate() -> bool:
+        return deployment_is_available(pytest.helm_release + "-csi-controller", pytest.helm_namespace)
+
+    assert retry_predicate(predicate, pytest.max_retries, pytest.retry_seconds)
```

Comment:
Nit: You can also use pytest modules to retry. Something like this on top of the method `@retry(wait_fixed=100, stop_max_attempt_number=50)`

Reply:
Resolving this conversation, because this test is no longer a part of this PR.